frame/contracts/proc-macro/src/lib.rs

```diff
@@ -167,8 +165,20 @@ enum HostFnReturn {
 	ReturnCode,
 }
 
+impl HostFnReturn {
+	fn to_wasm_sig(&self) -> TokenStream2 {
+		let ok = match self {
+			Self::Unit => quote! { () },
+			Self::U32 | Self::ReturnCode => quote! { ::core::primitive::u32 },
+		};
+		quote! {
+			Result<#ok, ::wasmi::core::Trap>
```

Comment:
I am not sure how "correct" you want to be for this proc. macro since its only user is yourself and therefore it probably isn't too important but there is some space for better macro hygiene. For example non-prelude names for all used identifiers. I usually do this even for macros where this is not important so that it becomes a muscle reflex to do this properly for all macros.